src/Models/DeletedModel.php

```diff
@@ -63,6 +67,22 @@ public function restore(Closure $beforeSaving = null): Model
         return $restoredModel;
     }
 
+    protected function enableIdentityInsert(string $modelClass)
+    {
+        if (config('database.default') === 'sqlsrv') {
```

Comment:
What happens if the model doesn't use the default database?

Reply:
Thank you for the observation. A new commit is summitted.